src/js/aframe/aframeForcegraph.js

```diff
@@ -0,0 +1,8816 @@
+	    lineOpacity: {type: 'number', default: 0.2},
+	    autoColorBy: {type: 'string'}, // color nodes with the same field equally
+	    idField: {type: 'string', default: 'id'},
+	    valField: {type: 'string', default: 'val'},
+	    nameField: {type: 'string', default: 'name'},
+	    colorField: {type: 'string', default: 'color'},
+	    linkSourceField: {type: 'string', default: 'source'},
+	    linkTargetField: {type: 'string', default: 'target'},
+	    forceEngine: {type: 'string', default: 'd3'}, // 'd3' or 'ngraph'
+	    warmupTicks: {type: 'int', default: 0}, // how many times to tick the force engine at init before starting to render
+	    cooldownTicks: {type: 'int', default: Infinity},
+	    cooldownTime: {type: 'int', default: 15000} // ms
+	  },
+
+	  init: function () {
+			this.state = {}; // Internal state
+      // Get camera dom element
+	    var cameraEl = document.querySelector('a-entity[camera], a-camera');
+
+	    // Add info msg
+	    cameraEl.appendChild(this.state.infoEl = document.createElement('a-text'));
+	    this.state.infoEl.setAttribute('position', '0 -0.1 -1'); // Canvas center
+	    this.state.infoEl.setAttribute('width', 1);
+	    this.state.infoEl.setAttribute('align', 'center');
+	    this.state.infoEl.setAttribute('color', 'lavender');
+
+	    // Setup tooltip (attached to camera)
+	    cameraEl.appendChild(this.state.tooltipEl = document.createElement('a-text'));
+	    this.state.tooltipEl.setAttribute('position', '0 -0.5 -1'); // Aligned to canvas bottom
+	    this.state.tooltipEl.setAttribute('width', 2);
+	    this.state.tooltipEl.setAttribute('align', 'center');
+	    this.state.tooltipEl.setAttribute('color', 'lavender');
+	    this.state.tooltipEl.setAttribute('value', '');
+
+	    // Keep reference to Three camera object
+	    this.cameraObj = cameraEl.object3D.children
+	        .filter(function(child) { return child.type === 'PerspectiveCamera' })[0];
+
+	    // Add D3 force-directed layout
+	    this.state.d3ForceLayout = d3.forceSimulation()
+	        .force('link', d3.forceLink())
+	        .force('charge', d3.forceManyBody())
+	        .force('center', d3.forceCenter())
+	        .stop();
+
+		},
+
+	  remove: function () {
+	    // Clean-up elems
+	    this.state.infoEl.remove();
+	    this.state.tooltipEl.remove();
+	  },
+
+	  update: async function (oldData) {
+	    var comp = this,
+	        elData = this.data,
+	        diff = AFRAME.utils.diff(elData, oldData);
+
+	    comp.state.onFrame = null; // Pause simulation
+	    comp.state.infoEl.setAttribute('value', 'Loading...'); // Add loading msg
+
+			/* if ('jsonUrl' in diff && elData.jsonUrl) {
+	      // (Re-)load data
+	      qwest.get(elData.jsonUrl).then(function(_, json) {
+	        elData.nodes = json.nodes;
+	        elData.links = json.links;
+
+	        comp.update(elData);  // Force re-update
+	      });
+	    } */
+			// Auto add color to uncolored nodes
+			const { nodes, links } = await aframeHelper.getData();
```

Comment:
@jonathanKingston as a quick hack, I decided to pass `nodes, links` into this aframe-component. 

Because we weren't using a local server, the above `qwest` code didn't work for json file. The `qwest` library uses `XMLHTTPRequest` by default. 

I shall think about submitting a patch for this aframe-component itself :)